audio-worklet/test/bitcrusher-test.js

```diff
@@ -0,0 +1,104 @@
+copyright notice, this list of conditions and the following disclaimer
+in the documentation and/or other materials provided with the
+distribution.
+  * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+//...  Populate array with 0:99 and verify the bitcrusher holds [0, 0, ... 0,
+//...  10, 10, ..., 10] where the number of consecutive equal numbers is  
+//...  controlled by the variable factor. 
+//...  Because of floating precision error this test fails to be precise
+//...  if the phaserInc variable in bitcrusher.js (1/factor) is not greater or 
+//...  equal to 1 when multiplied by factor. So stop test at 5
+let offlineContext = new OfflineAudioContext(2, 48000, 48000); 
+let bitcrusher = new Bitcrusher(offlineContext, {});
+
+for (let factor = 1; factor < 5; factor++) {
+  const precision = 1; 
+  const bufferSize = 100; 
+  let referenceBuffer = new Float32Array(bufferSize);
+  let outputBuffer = new Float32Array(bufferSize);
+  
+  // Populate array 1:100
+  for (let i =0; i < bufferSize; i++) {
+    referenceBuffer[i] = i; 
+  }
+  bitcrusher.processBuffer_(factor, precision, referenceBuffer, outputBuffer); 
+
+  // Verify computed values match expected values
+  outputBuffer.map((data, index) => { 
+    let expected = Math.floor(index / factor) * factor;
+    console.assert(data == expected, "computed " + data +
+     " but expected " + expected);
+  });
+}
+
+//... Create two oscillators, runing one through a bitcrusher with variables set  
+//... to avoid any effect. Conjoin nodes through a ChannelMergerNode 
+//... and feed output to an offline destination. Then verify that the samples 
+//... are nearly identical
+reduction = 1; 
+bitDepth = 24; 
+let osc = new OscillatorNode(offlineContext); 
+let oscb = new OscillatorNode(offlineContext); 
+const bufferLength = 512;
+osc.start();  
+oscb.start();
+
+let merger = new ChannelMergerNode(offlineContext, {numberOfInputs: 2}); 
+let bc = new Bitcrusher(offlineContext, {buffersize: 512, inputChannels: 1, 
+  outputChannels: 1, bitDepth, reduction});
```

Comment:
Actually, this works since the variable names match up (ie, the expression gets expanded to bitDepth: bitDepth, reduction:reduction.)